app/pulp/app/models/progress.py

```diff
@@ -18,6 +20,16 @@ class ProgressReport(Model):
 
     All progress reports have a message, state, and are related to a Task.
 
+    Attributes:
+
+        batch_increments (bool): If this Flag is set save checks to make sure it only gets ran once
```

Comment:
Fair enough. I think defaulting to giving plugin writers more power is good, but I also see how we would not want to overwhelm them with too many options. (and this is something easily added if someone ever requests this feature)
I will amend this PR.